frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.ts

```diff
@@ -2,23 +2,37 @@ import { PipelineRunKFv2, PipelineSpecVariable } from '~/concepts/pipelines/kfTy
 import { createNode } from '~/concepts/topology';
 import { PipelineNodeModelExpanded } from '~/concepts/topology/types';
 import { createArtifactNode } from '~/concepts/topology/utils';
+import { usePipelineRunMlmdContext } from '~/concepts/pipelines/apiHooks/mlmd/usePipelineRunMlmdContext';
+import { useExecutionsFromContext } from '~/concepts/pipelines/apiHooks/mlmd/useExecutionsFromContext';
+import { isPipelineRunComplete } from '~/concepts/pipelines/apiHooks/usePipelineRunById';
 import {
   composeArtifactType,
   parseComponentsForArtifactRelationship,
   parseInputOutput,
-  parseRuntimeInfo,
+  parseRuntimeInfoFromExecutions,
+  parseRuntimeInfoFromRunDetails,
   parseTasksForArtifactRelationship,
   parseVolumeMounts,
   translateStatusForNode,
 } from './parseUtils';
 import { KubeFlowTaskTopology } from './pipelineTaskTypes';
 
-const EMPTY_STATE: KubeFlowTaskTopology = { taskMap: {}, nodes: [] };
+const EMPTY_STATE: KubeFlowTaskTopology = { taskMap: {}, nodes: [], loaded: true };
 
 export const usePipelineTaskTopology = (
   spec?: PipelineSpecVariable,
   run?: PipelineRunKFv2,
 ): KubeFlowTaskTopology => {
+  const isComplete = isPipelineRunComplete(run);
```

Comment:
nit: The concepts of "complete" and "finished" are similar, but maybe just stick with 1 or the other across the board.

Reply:
Complete can sometimes be associated with a success, so maybe "finished" would be more appropriate.